### PR TITLE
Add Operator guide

### PR DESCRIPTION
**Pull request by ahardin-rh**
Preview Build: http://file.rdu.redhat.com/~ahardin/012020/operator-guide-reference/operators/operator-reference.html

Per an email discussion with @sferich888 on January 20, if the GitHub links align with what we see in `oc adm release info --commits IMAGE`, then the usage of GitHub links to to each project is okay.

### CODE REVIEW

**Comment by jboxman** [comment]
At some point, it would be interesting to include the definitions from https://try-apidocs--ocpdocs.netlify.com/openshift-enterprise/latest/rest_api/index.html#alertmanager-v1-monitoring-coreos-com. But the content isn't modularized. But I did envision in some future that we'll be able to do that, possibly.

@vikram-redhat, fyi

**Comment by ahardin-rh** [comment]
@rjhowe Getting some initial content into place. What do you think?

@cuppett PTAL at my first pass as well.

**Comment by ahardin-rh** [comment]
@sferich888 PTAL; as the content grows, we may break this content into subsections, but we're starting with one file for now.

**Comment by ahardin-rh** [comment]
@emmajiafan Can you please provide a QE review or point me to the correct contact? Thank you!

File: modules/cluster-monitoring-operator.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+** Scope: Namespaced
+** CR: prometheus
+** Validation: Yes
+* prometheusrules.monitoring.coreos.com
+** Scope: Namespaced
+** CR: prometheusrule
+** Validation: Yes
+* servicemonitors.monitoring.coreos.com
+** Scope: Namespaced
+** CR: prometheusrule
```
Comment (juzhao) [inline modules/cluster-monitoring-operator.adoc]: CR: servicemonitor

File: modules/cluster-monitoring-operator.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+** Validation: Yes
+
+[discrete]
+== Configuration objects
+
+----
+$ oc edit  cm cluster-monitoring-config
```
Comment (juzhao) [inline modules/cluster-monitoring-operator.adoc]: oc -n openshift-monitoring edit cm cluster-monitoring-config

File: modules/prometheus-operator.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+persistence, retention policies, and replicas from a native Kubernetes resource.
+
+* Target Services via Labels: Automatically generate monitoring target
+configurations based on familiar Kubernetes label queries; no need to learn a
+Prometheus specific configuration language.
+
+[discrete]
+== Project
+
+link:https://github.com/openshift/prometheus-operator[prometheus-operator]
```
Comment (juzhao) [inline modules/prometheus-operator.adoc]: I think we should move the project link to the top of the page, I see the project link for "Cluster Monitoring Operator" is at the top of the page, see https://github.com/openshift/openshift-docs/blob/d4476fe76420e1f9da4eaed573b790391d75b8a7/modules/cluster-monitoring-operator.adoc

File: modules/kube-apiserver-operator.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+// * operators/operator-reference.adoc
+
+[id="kube-apiserver-operator_{context}"]
+= Kubernetes API Server Operator
+
+[discrete]
+== Purpose
+
+The Kubernetes API Server Operator manages and updates the Kubernetes API server
+deployed on top of {product-title}.
```
Comment (wangke19) [inline modules/kube-apiserver-operator.adoc]: Refer to the README, add 'The operator is based on OpenShift library-go framework and it is installed via Cluster Version Operator (CVO)' to the end of the line.

File: modules/console-operator.adoc [hunk 5]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
```
Comment (yapei) [inline modules/console-operator.adoc]: console-operator parts look good

File: modules/cluster-image-registry-operator.adoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+resource status with information about what is missing.
+
+The Cluster Image Registry Operator runs in the `openshift-image-registry`
+namespace and it also manages the registry instance in that location. All
+configuration and workload resources for the registry reside in that namespace.
+
+[discrete]
+== Project
+
+link:https://github.com/openshift/cluster-image-registry-operator[cluster-image-registry-operator]
```
Comment (wzheng1) [inline modules/cluster-image-registry-operator.adoc]: Acked by QE

File: modules/cluster-openshift-controller-manager-operators.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+can be viewed in a cluster with:
+
+----
+$ oc get crd openshiftcontrollermanagers.operator.openshift.io -o yaml
+----
+
+[discrete]
+== Project
+
+link:https://github.com/openshift/cluster-openshift-controller-manager-operator[cluster-openshift-controller-manager-operator]
```
Comment (wzheng1) [inline modules/cluster-openshift-controller-manager-operators.adoc]: Acked by QE.

File: modules/cluster-image-registry-operator.adoc [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+[discrete]
+== Purpose
+
+The Cluster Image Registry Operator manages a singleton instance of the
+{product-title} registry. It manages all configuration of the registry,
+including creating storage.
+
+On initial start up, the Operator creates a default `image-registry` resource
+instance based on the configuration detected. in the cluster. This indicates
+what cloud storage type to use based on the cloud provider
```
Comment (wzheng1) [inline modules/cluster-image-registry-operator.adoc]: Lack of "."

File: modules/cluster-samples-operator.adoc [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+//
+// *  operators/operator-reference.adoc
+
+[id="cluster-samples-operator_{context}"]
+= Cluster Samples Operator
+
+[discrete]
+== Purpose
+
+The Cluster Samples Operator manages the sample image streams and templates
```
Comment (xiuwang) [inline modules/cluster-samples-operator.adoc]: s/image stream/imagestream

File: modules/cluster-samples-operator.adoc [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+stored in the `openshift` namespace, and any container credentials, stored as a
+secret, needed for the image streams to import the images they reference.
```
Comment (xiuwang) [inline modules/cluster-samples-operator.adoc]: Same as above, need replace all `image stream` to `imagestream` in this doc

File: modules/cluster-samples-operator.adoc [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+On initial start up, the Operator creates the default samples resource to
+initiate the creation of the image streams and templates. The image streams are
+the {op-system-first}-based {product-title} image streams pointing to images on
+`registry.redhat.access.com`. Similarly, the templates are those categorized as
```
Comment (xiuwang) [inline modules/cluster-samples-operator.adoc]: the default registry is `registry.redhat.io` ,not `registry.redhat.access.com`

File: modules/openshift-apiserver-operator.adoc [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+== Project
+
+link:https://github.com/openshift/cluster-openshift-apiserver-operator[openshift-apiserver-operator]
+
+[discrete]
+== CRDs
+
+* openshiftapiservers.operator.openshift.io
+** Scope: Cluster
+** CR: openshiftapiserver
```
Comment (xingxingxia) [inline modules/openshift-apiserver-operator.adoc]: This part about my owned openshift-apiserver operator LGTM, thanks

File: modules/prometheus-operator.adoc (hunk 3 above)
Comment (ahardin-rh) [reply modules/prometheus-operator.adoc]: This page's Purpose statement is just longer. Thanks!

File: modules/cluster-openshift-controller-manager-operators.adoc (hunk 7 above)
Comment (ahardin-rh) [reply modules/cluster-openshift-controller-manager-operators.adoc]: Thank you! The content here will be built out soon, likely in a separate PR

**Review by lamek — COMMENTED** [review]
Called out some very minor things to correct or take a second look at. LGTM.

File: modules/operator-marketplace.adoc [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+[discrete]
+== Purpose
+
+The Marketplace Operator is a conduit to bring off-cluster Operators to your
+cluster.
+
+[discrete]
+== Project
+
+link:https://github.com/operator-framework/operator-marketplace[operator-marketplace]
```
Comment (emmajiafan) [inline modules/operator-marketplace.adoc]: This part about marketplace LGTM, thanks

File: modules/cluster-svcat-apiserver-operator.adoc [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+comprised of an aggregated API server and a controller manager; this Operator
+only deals with the API server portion of Service Catalog. See
+link:https://github.com/openshift/cluster-svcat-controller-manager-operator[cluster-svcat-controller-manager-operator]
+for the Operator responsible for the controller manager component of Service
+Catalog.
+
+[discrete]
+== Project
+
+link:https://github.com/openshift/cluster-svcat-apiserver-operator[cluster-svcat-apiserver-operator]
```
Comment (emmajiafan) [inline modules/cluster-svcat-apiserver-operator.adoc]: This part LGTM, thanks

File: modules/cluster-svcat-controller-manager-operator.adoc [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+instance of the OpenShift Service Catalog on a cluster. Service Catalog is
+comprised of an aggregated API server and a controller manager; this Operator
+only deals with the Controller Manager portion of Service Catalog. See the
+link:https://github.com/openshift/cluster-svcat-apiserver-operator[cluster-svcat-apiserver-operator]
+for the Operator responsible for the API Server component of Service Catalog.
+
+[discrete]
+== Project
+
+link:https://github.com/openshift/cluster-svcat-controller-manager-operator[cluster-svcat-controller-manager-operator]
```
Comment (emmajiafan) [inline modules/cluster-svcat-controller-manager-operator.adoc]: This part LGTM, thanks.

**Pull request merged**